src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetCore.cs

```diff
@@ -53,12 +53,14 @@ private static Assembly AssemblyResolve_ManagedProfilerDependencies(object sende
 
             // Only load the main profiler into the default Assembly Load Context.
             // If Datadog.Trace or other libraries are provided by the NuGet package their loads are handled in the following two ways.
-            // 1) The AssemblyVersion is greater than or equal to the version used by Datadog.Trace.ClrProfiler.Managed, the assembly
+            // 1) The AssemblyVersion is greater than or equal to the version used by Datadog.Trace, the assembly
             //    will load successfully and will not invoke this resolve event.
-            // 2) The AssemblyVersion is lower than the version used by Datadog.Trace.ClrProfiler.Managed, the assembly will fail to load
+            // 2) The AssemblyVersion is lower than the version used by Datadog.Trace, the assembly will fail to load
             //    and invoke this resolve event. It must be loaded in a separate AssemblyLoadContext since the application will only
             //    load the originally referenced version
-            if (assemblyName.Name.StartsWith("Datadog.Trace.ClrProfiler.Managed", StringComparison.OrdinalIgnoreCase)
+
+            // TODO: confirm that this check is now correct since StartsWith("Datadog.Trace") will match more assembly names
```

Comment:
TODO: confirm that this check is now correct since StartsWith("Datadog.Trace") will match more assembly names

Reply:
@zacharycmontoya: I think to match previous behavior with regards to `Datadog.Trace.dll`, we should remove the first `if` clause entirely and only leave:
```csharp
if (File.Exists(path))
{
	StartupLogger.Debug("Loading {0} with DependencyLoadContext.LoadFromAssemblyPath", path);
	return DependencyLoadContext.LoadFromAssemblyPath(path); // Load unresolved framework and third-party dependencies into a custom Assembly Load Context
}
```
If previous versions, if an app included an older nuget package (for example, 1.28.0 home directory and 1.27.1 nuget), we would load both version of `Datadog.Trace.dll`, and all spans would show up, although automatic and manual spans would be split into two traces.

With `assemblyName.Name.StartsWith("Datadog.Trace")`, we fail to load the new dll, so only manual spans show up.

WDYT?

Reply:
(I made the change and waiting for the build to test it.)

Reply:
Confirmed it now behaves like before. Regression fixed.

Reply:
Yeah sounds good to me, if the user provides their own `Datadog.Trace.dll`, then we must load our own into a separate ALC 👍🏼